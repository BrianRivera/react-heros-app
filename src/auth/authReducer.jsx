const { types } = require("../types/types");

const state = {
    name: 'Brina',
    logged: true
}

export const authReducer = (state = {}, action) => {

    switch (action.type) {
        case types.login:
            return {
                ...action.payload,
                logged:true
            }
        
            case types.logaut:
                return {
                    logged: false
                }
    
        default:
            return state;
            break;
    }
    
}
