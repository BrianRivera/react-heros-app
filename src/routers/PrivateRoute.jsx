import React from 'react'
import { Route, Redirect } from 'react-router-dom'
import PropTypes from 'prop-types';


export const PrivateRoute = ({
    isAutenticated,
    component: Component,
    ...rest
}) => {


    localStorage.setItem('lastPath',rest.location.pathname)

    return (
        <Route
        {...rest}
        component = {(props) =>(
            (isAutenticated)
            ? <Component {...props}></Component>
            : (<Redirect to="/login"></Redirect>)
        )}
        >


        </Route>
    )
}

PrivateRoute.propTypes = {
    isAutenticated: PropTypes.bool.isRequired,
    component: PropTypes.func.isRequired
}