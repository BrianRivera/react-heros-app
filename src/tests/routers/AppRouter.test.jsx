import '@testing-library/jest-dom';
import React from 'react';
import { shallow, mount } from 'enzyme';
import { AppRouter } from '../../routers/AppRouter';
import { AuthContext } from '../../auth/AuthContext';

describe('Pruebas en AppRouter', () => {

    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged:false
        }
    }

    test('debe de mostrar el login si no esta autenticado', () => {
        

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
            <AppRouter></AppRouter>
            </AuthContext.Provider>
            )
            expect(wrapper).toMatchSnapshot();

    });

    test('should debe mostrar el componente de marvel si esta autenticado', () => {

        const contextValue = {
            dispatch: jest.fn(),
            user: {
                logged:true,
                name:'brian'
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value={contextValue}>
            <AppRouter></AppRouter>
            </AuthContext.Provider>
            )
            expect(wrapper.find('.navbar').exists()).toMatchSnapshot();
            
    })
    
    
    
})
