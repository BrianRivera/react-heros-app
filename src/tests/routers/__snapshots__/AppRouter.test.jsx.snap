// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en AppRouter debe de mostrar el login si no esta autenticado 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      login
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      login
    </button>
  </div>
</div>
`;

exports[`Pruebas en AppRouter should debe mostrar el componente de marvel si esta autenticado 1`] = `true`;
